lib/spack/spack/reporters/cdash.py

```diff
@@ -239,13 +233,13 @@ def clean_log_event(event):
                 f.write(t.render(report_data))
             self.upload(phase_report)
 
-    def build_report(self, directory_name, input_data):
+    def build_report(self, directory_name, specs):
```

Comment:
This is another part of the API that bugged me. The base class was accepting a `filename` (it started with `JUnit` which produces a single file). To accomodate CDash the string became a `directory_name`. Should we just call this "location" in the base class and document that the meaning of it varies according to the specific implementation?

Reply:
I think it's either that, or change JUnit to generate the single file in a directory.  I kind of like the latter better, as it means that every reporter gets its own scratch directory to work in.  But if JUnit only needs one file maybe it's overkill.